src/ILCompiler.Compiler/src/IL/ILImporter.Scanner.cs

```diff
@@ -406,6 +406,12 @@ private void ImportCall(ILOpcode opcode, int token)
 
             TypeDesc exactType = method.OwningType;
 
+            if ((method.Signature.IsStatic && opcode == ILOpcode.callvirt)
```

Comment:
Could you also separate it out here (or just remove the check for static method with callvirt)? The structure of the scanner mirrors the structure in CorInfoImpl.cs whenever possible.

Also, after looking at this several times, maybe we should use `ThrowHelper.ThrowInvalidProgramException()` for the native callable case. Might be more appropriate, after re-reading MSDN on those two exceptions.

Looks good otherwise.